### PR TITLE
v2: allow disabling MSI

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/3278

### CODE REVIEW

**Comment by xh3b4sd** [comment]
How can we verify this is what certain people ask for?

**Comment by kopiczko** [comment]
@r7vme can you try to run this on german cloud?

**Comment by r7vme** [comment]
Yep, sure. Will pick up as soon as will have a time. Anyway wrote in slack to @kopiczko 

In general, i assume this change only temporary to test this use case and then we will make disabled MSI optional.

**Comment by r7vme** [comment]
@kopiczko your change works fine in German cloud. Do you have ideas how we can optionally disable MSI?

**Comment by kopiczko** [comment]
> Do you have ideas how we can optionally disable MSI?
Let's see if I can do that easily.

File: service/controller/v2/arm_templates/vmss.json [hunk 1]
```diff
@@ -70,6 +73,21 @@
...
+                    "type": "ManagedIdentityExtensionForLinux",
+                    "typeHandlerVersion": "1.0",
+                    "autoUpgradeMinorVersion": true,
+                    "settings": {
+                        "port": 50342
+                    },
+                    "protectedSettings": {}
+                }
+            }
+        ],
```
Comment (r7vme) [inline service/controller/v2/arm_templates/vmss.json]: You also need to disable roleAssignments. You can use condition like here

https://github.com/giantswarm/azure-operator/blob/master/service/controller/v2/arm_templates/route_table_setup.json#L37

Comment (kopiczko) [reply service/controller/v2/arm_templates/vmss.json]: It's done already: https://github.com/giantswarm/azure-operator/pull/245/files#diff-b64f09db2486367b61a55095d8016d16R278

**Comment by kopiczko** [comment]
@r7vme e2e tests fails because of failing deployment:

```
{"code":"DeploymentFailed","message":"At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-debug for usage details.","details":[{"code":"BadRequest","message":"{\r\n \"error\": {\r\n \"code\": \"InvalidResourceReference\",\r\n \"message\": \"Resource /subscriptions/1be3b2e6-497b-45b9-915f-eb35cae23c6a/resourceGroups/CI-WIP-EC318-5E958/providers/Microsoft.Network/publicIPAddresses/DUMMY-PIP referenced by resource /subscriptions/1be3b2e6-497b-45b9-915f-eb35cae23c6a/resourceGroups/ci-wip-ec318-5e958/providers/Microsoft.Network/loadBalancers/kubernetes was not found. Please make sure that the referenced resource exists, and that both resources are in the same region.\",\r\n \"details\": [\r\n {\r\n \"code\": \"NotFound\",\r\n \"message\": \"Resource /subscriptions/1be3b2e6-497b-45b9-915f-eb35cae23c6a/resourceGroups/CI-WIP-EC318-5E958/providers/Microsoft.Network/publicIPAddresses/DUMMY-PIP not found.\"\r\n }\r\n ]\r\n }\r\n}"}]}
```

**Pull request merged**